pkg/cloud/meta/method.go

```diff
@@ -234,6 +234,10 @@ func (m *Method) init() {
 			if itemsType.Kind() != reflect.Slice && itemsType.Elem().Kind() != reflect.Ptr {
 				panic(fmt.Errorf("method %q.%q: paged return type %q.Items is not an array of pointers", m.Service, m.Name(), listType.Name()))
 			}
+
+			if itemsType.Kind() != reflect.Slice || itemsType.Elem().Kind() != reflect.Pointer {
```

Comment:
the previous check and this are confusing, first the names Ptr vs Pointer, it seems Ptr was deprecated and now means the same as Pointer https://pkg.go.dev/reflect#pkg-constants and https://github.com/golang/go/issues/47651

if Ptr and Pointer is the same this is just the previous check with an OR instead of an AND , so this is more restrictive than the previous that requires both to be different and now, we panic if at least one is different

Should you remove the previous check then?

Reply:
ok, looks like the real bug was a conditional that was wrong... fixing